### PR TITLE
fix: add `-z noexecstack` in Makevars

### PR DESCRIPTION
**Pull request by etiennebacher**
Error seen in ubuntu CI workflow of tidypolars

### CODE REVIEW

File: src/Makevars.in
Comment (CGMossa) [inline src/Makevars.in:1]: Hello! The flag you found goes to the linker, not `cargo` itself. The location should be somewhere that gets passed to _R's build step_ of the package, in which the linker is invoked (oddly enough, the linker interface is `gcc` too lol). I think you can start by putting this in `PKG_LIBS` line and see if it helps.

Comment (etiennebacher) [reply src/Makevars.in:1]: Thanks, let's see how it goes

Comment (etiennebacher) [reply src/Makevars.in:1]: @CGMossa it seems that this flag only exists on Linux (just my guess since ubuntu passes but macos workflows fail with `ld: unknown options: -z `

Comment (etiennebacher) [reply src/Makevars.in:1]: My bad, I *think* this option should go in the configure file, no makevars.in

**Comment by CGMossa** [comment]
This is so good to know.... I'm wondering if this needs to be upstreamed to extendr.

**Comment by etiennebacher** [comment]
I still don't know if this worked, I need to wait for 0.19.1 to be available on R-universe. Also, this is the first time I see this, I never had an glibc issue in CI before. But you know more than me what should and shouldn't be handled in extendr (might be part of a FAQ / "good to know" section in the docs.

File: DESCRIPTION [hunk 1]
```diff
@@ -118,5 +118,5 @@ Collate:
     'zzz.R'
 Config/rextendr/version: 0.3.1
 VignetteBuilder: knitr
-Config/polars/LibVersion: 0.42.0
+Config/polars/LibVersion: 0.42.1
```
Comment (eitsupi) [inline DESCRIPTION:121]: Was this change necessary?
Isn't this flag unrelated to the Rust binary?

Comment (etiennebacher) [reply DESCRIPTION:121]: Maybe it wasn't, I think I misinterpreted the failures in the first commit because I've been bitten before by compilation errors when I forgot to update the libversion
